polaris-react/src/components/IndexFilters/IndexFilters.tsx

```diff
@@ -95,6 +104,12 @@ export interface IndexFiltersProps
   filteringAccessibilityTooltip?: string;
 }
 
+export type IndexFiltersProps =
+  | ((IndexFiltersPropsBase & SearchFilterButtonPropsHiddenQuery) &
+      Never<SearchFilterButtonPropsPeekQuery>)
+  | ((IndexFiltersPropsBase & SearchFilterButtonPropsPeekQuery) &
+      Never<SearchFilterButtonPropsHiddenQuery>);
```

Comment:
Mutually exclusive props (can't have `peekQueryField` and `hideQueryField` be booleans at the same time)